### DISCUSSION TITLE
Cashier Stripe pricing table not working

### DISCUSSION THREAD

**Discussion by ITrunsDE** [laravel/framework#48163] [Q&A]
Is there a way to make the pricing table feature of Stripe usable in Laravel?

https://stripe.com/docs/payments/checkout/pricing-table

When entering the page, the authenticated customer is created in Stripe. But when clicking on one of the displayed prices, no transmission of the customer ID takes place. And in the Stripe portal another customer is created. 

There is a method https://stripe.com/docs/payments/checkout/pricing-table#customer-session. 

I have tried via 

```PHP
$intent = auth()->user()->createSetupIntent();
```

and then in the JS code

```Javascript
...
  customer-session-client-secret="{{ $intent->client_secret }}"
```

After making the payment, I have a new customer in the Stripe portal. But the existing customer in Laravel Project has no subscription.

Does anyone have any idea?

Thank you very much in advance.
Sebastian

**Comment by andrei930** [comment]
Hi, 

From what I see, if you use the pricing tables flow, Stripe creates a new customer and only passes back the client-reference-id in the checkout.session.completed event. All other webhook requests will be sent using Stripes created customer id, not the one associated with your user.

**Comment by drsdre** [comment]
I've done a Stripe pricing table integration like this:
```
        $session = Cashier::stripe()->customerSessions->create([
            'customer' => $tenant->stripeId(),
            'components' => [
                'pricing_table' => ['enabled' => true],
            ],
        ]);
```

Once the session is established, the pricing table is generated like this:
```
    <script async src="https://js.stripe.com/v3/pricing-table.js"></script>
    <stripe-pricing-table pricing-table-id="{{$pricing_table_id}}"
                          publishable-key="{{config( 'cashier.key')}}"
                          customer-session-client-secret="{{$session->client_secret}}"
    >
    </stripe-pricing-table>
```

However in my tests, I found issues like your and also an issue where Stripe tells te customer to confirm his email address on checkout even though the customer has full access to the Stripe customer portal. It seems that the user session is indeed not getting established as intended.
